pyproject.toml

```diff
@@ -1,18 +1,56 @@
 [build-system]
 requires = [
-    "cython==3.0.8",
-    "setuptools==71.1.0",
-    "setuptools_dso==2.10;platform_system != 'Windows'",  # On Windows we don't distribute a dynamic library
-    "wheel>=0.30.0",
+    "cython~=3.0",
+    "meson-python~=0.16",
 ]
-build-backend = "setuptools.build_meta"
+build-backend = 'mesonpy'
+
+[project]
+name = "dwave-optimization"
+dynamic = ["version"]
+authors = [
+    {name = "D-Wave Inc.", email = "tools@dwavesys.com"},
+]
+description = "Enables the formulation of nonlinear models for industrial optimization problems."
+license = {file = "LICENSE"}
+classifiers = [
+    "License :: OSI Approved :: Apache Software License",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX",
+    "Operating System :: Unix",
+    "Operating System :: MacOS",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Programming Language :: Python :: 3 :: Only",
+]
+requires-python = ">=3.8"
+dependencies = [
+    "numpy>=1.20.0",  # numpy.typing support
+]
+
+[project.readme]
+file = "README.rst"
+content-type = "text/x-rst"
+
+[project.urls]
+Changelog = "https://github.com/dwavesystems/dwave-optimization/releases"
+Documentation = "https://docs.ocean.dwavesys.com"
+"Source Code" = "https://github.com/dwavesystems/dwave-optimization"
```

Comment:
Why these url names? I've used e.g., [this](https://daniel.feldroy.com/posts/2023-08-pypi-project-urls-cheatsheet) or [this](https://github.com/patrick91/links-demo) as references.

Reply:
:shrug: Why not these? There doesn't seem to be any consensus even within `pypa`

https://github.com/numpy/numpy/blob/ae926d7e01053baede90a989fb9bf98515d23a22/pyproject.toml#L53-L59
https://github.com/pypa/cibuildwheel/blob/3a30bec9ef7f8521c8f98e77a7091deafb3938a9/pyproject.toml#L91-L93
https://github.com/pypa/setuptools/blob/8e963821d37216f6ba416b14a51c628606be7f76/pyproject.toml#L31-L33
https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#urls

These are the ones I like to see so went with them. I am not sure this is a place where consistency matters much.

Reply:
Fair enough. I guess the only thing would be getting relevant icons for PyPI, but as you say it seems to work fine no matter what's used.